pkg/collector/python/test_check.go

```diff
@@ -188,16 +188,15 @@ void reset_check_mock() {
 import "C"
 
 func testRunCheck(t *testing.T) {
-	// Initialize rtloader pointer
-	rtloader = &C.rtloader_t{}
+	initMockRtLoaderPtr()
```

Comment:
Not a blocker:

When reading the code, it is not obvious why `defer func() { rtloader = nil }()` is required (You have to check the implementation of `initMockRtLoaderPtr`)
What do you think about one of these alternative solutions 
```go
rtloader = initMockRtLoaderPtr()
defer func() { rtloader = nil }()
```
Or
```go
cleanup := initMockRtLoaderPtr()  
defer cleanup()
``` 
`initMockRtLoaderPtr` returns a function that set `rtloader` to `nil`

Reply:
Yup I see what you mean - I'll make the change to match your first suggestion :)